harvard-lts/mirador-harvard-custom-plugin
# New mirador plugin for Harvard customizations

**New mirador plugin for Harvard customizations**

---

**JIRA Ticket**: [LTSVIEWER-266](https://jira.huit.harvard.edu/browse/LTSVIEWER-266)

# What does this Pull Request do?

Creates a new mirador plugin repository for Harvard customizations, which currently exist inside the mps-viewer repo

# How should this be tested?

Confirm it is still working in the plugin itself:
* Pull in the latest `mirador-harvard-custom-plugin` code from the `LTSVIEWER-266` branch.
* Run `npm run serve` to open up the demo site
* Confirm that the following elements are present:
   * Harvard shield in the lower left corner (this is just an image, not a link)
   * If v2 or v3 manifest: The link underneath "Attribution" or "Provided by" (Toggle sidebar -> Rights) reads as normal text, "Copyright & Terms of Use" rather than displaying the url itself
   *  If legacy manifest: The link underneath "License" (Toggle sidebar -> Rights) reads as normal text, "Copyright & Terms of Use" rather than displaying the url itself
   * Note: I put three manifest ids in the demoEntry.js file, so just comment/uncomment them out to see items with different labels ("Attribution", "Provided by", "License") that all change the copyright link into plain text

Confirm it is working in mps-viewer:
* In your local version of `mirador-harvard-custom-plugin`, run `npm run build` - This will create a new index.js file in the `/dist/es` folder
* Copy that `index.js` file into a new folder `/plugins/custom` in your local version of `mps-viewer`
* In `/src/mirador.js`, add a new line to import this code: `import miradorHarvardCustomPlugin from "../plugins/custom/index";` and add `miradorHarvardCustomPlugin` to the plugin list at the bottom of the file (~line 364)
* Remove imports and any mentions of `copyrightLabel` or `harvardBranding` (these were the previous local plugins driving the changes that now occur via this new plugin)
* Rebuild your local `mps-embed` and `mps-viewer` containers
* Confirm that the copyright label and harvard logo are showing up properly for all examples in `mps-viewer` (would be good to check one of each manifest type, legacy/v2/v3).

Once this is confirmed to be working, I will tag and deploy a new release of `mirador-harvard-custom-plugin` to NPM. Then I will create a new PR for `mps-viewer` that will use the new `mirador-harvard-custom-plugin` release.

# Test coverage

Yes/No: Are changes in this pull-request covered by:

- unit tests? no
- integration tests? no

# Interested parties
@enriquediaz
